enterprise/coderd/prebuilds/metricscollector.go

```diff
@@ -247,3 +271,20 @@ func (mc *MetricsCollector) trackResourceReplacement(orgName, templateName, pres
 	// cause an issue (or indeed if either would), so we just track the replacement.
 	mc.replacementsCounter[key]++
 }
+
+type hardLimitedPresetKey struct {
+	orgName, templateName, presetName string
+}
+
+func (k hardLimitedPresetKey) String() string {
+	return fmt.Sprintf("%s:%s:%s", k.orgName, k.templateName, k.presetName)
+}
+
+func (mc *MetricsCollector) trackHardLimitedStatus(orgName, templateName, presetName string, isHardLimited bool) {
+	mc.isPresetHardLimitedMu.Lock()
+	defer mc.isPresetHardLimitedMu.Unlock()
+
+	key := hardLimitedPresetKey{orgName: orgName, templateName: templateName, presetName: presetName}
+
+	mc.isPresetHardLimited[key] = isHardLimited
```

Comment:
nit: With this approach, we’re accumulating entries that are no longer relevant. This can cause the metrics to keep reporting `0` for presets that should disappear (e.g., deleted or outdated presets).

A small improvement would be to remove presets from the map when `isHardLimited == false`:
```
if isHardLimited {
	mc.isPresetHardLimited[key] = true
} else {
	delete(mc.isPresetHardLimited, key)
}
```

Reply:
Good catch, fixed here: https://github.com/coder/coder/pull/18008/commits/80c89fea87f181454af812f3df147f3ad4c42f42.

I was under a false assumption that it will report stale value indefinitely.